### PR TITLE
add doc for hostpath type

### PR DESCRIPTION
**Pull request by dixudx**
xref kubernetes/features#376

/cc @saad-ali @luxas

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5503)

### CODE REVIEW

**Comment by steveperry-53** [comment]
Please rebase your branch so that it picks up https://github.com/kubernetes/kubernetes.github.io/commit/99fbc2b4ae9a9a3c2fcb42d15337fa7efc0b36ca. That will fix the deploy/netlify failure. Thanks.

File: docs/concepts/storage/volumes.md [hunk 1]
```diff
@@ -141,22 +141,37 @@ spec:
...
 
 For example, some uses for a `hostPath` are:
 
 * running a container that needs access to Docker internals; use a `hostPath`
   of `/var/lib/docker`
 * running cAdvisor in a container; use a `hostPath` of `/dev/cgroups`
+* allowing a pod to specify whether a given hostPath should exist prior to the
+  pod running, whether it should be created, and what it should exist as
+
+Supported HostPath Types are:
```
Comment (tengqm) [inline docs/concepts/storage/volumes.md]: Change line 155 to this?

In addition to the required `path` property, user can optionally specify a `type` for a hostPath volume.
The supported values for field `type` are:

File: docs/concepts/storage/volumes.md [hunk 2]
```diff
@@ -141,22 +141,37 @@ spec:
...
+
+| Value | Behavior |
+|:------|:---------|
+| *unset* | For backwards compatible, leave it empty if unset. **Default value**. |
```
Comment (tengqm) [inline docs/concepts/storage/volumes.md]: change line 159 to this?

| `` | Empty string (default) or unset are for backward compatibility. It means that Kubernetes performs *no* checks before mounting the hostPath volume. |

**Comment by steveperry-53** [comment]
Ping @dixudx

File: docs/concepts/storage/volumes.md (hunk 1 above)
Comment (dixudx) [reply docs/concepts/storage/volumes.md]: @tengqm SGTM. Thanks.

**Comment by dixudx** [comment]
@steveperry-53  Updated. PTAL. Thanks.

File: docs/concepts/storage/volumes.md [hunk 3]
```diff
@@ -141,22 +141,40 @@ spec:
...
+  pod running, whether it should be created, and what it should exist as
+
+In addition to the required `path` property, user can optionally specify a `type` for a `hostPath` volume.
+
+The supported values for field `type` are:
+
+
+| Value | Behavior |
+|:------|:---------|
+| | Empty string (default) are for backward compatibility, which means that no checks will be performed before mounting the hostPath volume. |
```
Comment (tengqm) [inline docs/concepts/storage/volumes.md]: are for -> is for

File: docs/concepts/storage/volumes.md [hunk 4]
```diff
@@ -141,22 +141,40 @@ spec:
...
+| `DirectoryOrCreate` | If nothing exists at the given path, an empty directory will be created there as needed with file mode 0755, having the same group and ownership with Kubelet. |
```
Comment (tengqm) [inline docs/concepts/storage/volumes.md]: file mode -> permission set to

File: docs/concepts/storage/volumes.md [hunk 5]
```diff
@@ -141,22 +141,40 @@ spec:
...
+| `Directory` | A directory must exist at the given path |
+| `FileOrCreate` | If nothing exists at the given path, an empty file will be created there as needed with file mode 0644, having the same group and ownership with Kubelet. |
```
Comment (tengqm) [inline docs/concepts/storage/volumes.md]: file mode -> permission set to

**Comment by steveperry-53** [comment]
@dixudx @tengqm Docs LGTM. I'll merge this as soon as we have a tech approval.

**Pull request merged**